File: mobile/lib/constants/tipo_deficiencia_familiar.dart
enum TipoDeficienciaFamiliar {
  cegueira('CEGUEIRA', 'Cegueira'),
  baixaVisao('BAIXA_VISAO', 'Baixa Visão'),
  surdezLeveModerada('SURDEZ_LEVE_MODERADA', 'Surdez leve/moderada'),
  deficienciaFisica('DEFICIENCIA_FISICA', 'Deficiencia Fisica'),
  deficienciaMentalIntelectual(
    'DEFICIENCIA_MENTAL_INTELECTUAL',
    'Deficiencia Mental ou Intelectual',
  ),
  sindromeDown('SINDROME_DOWN', 'Sindrome de Down'),
  transtornoMental('TRANSTORNO_MENTAL', 'Transtorno Mental');

  const TipoDeficienciaFamiliar(this.apiValue, this.rotulo);

  final String apiValue;
  final String rotulo;

  static TipoDeficienciaFamiliar? fromApi(String? value) {
    if (value == null) return null;
    for (final t in values) {
      if (t.apiValue == value) return t;
    }
    return null;
  }
}

File: backend/src/lib/tipo-deficiencia-familiar.ts
import { TipoDeficienciaFamiliar } from "@prisma/client";

export type TipoDeficienciaFamiliarDef = {
  valor: TipoDeficienciaFamiliar;
  rotulo: string;
};

export const TIPOS_DEFICIENCIA_FAMILIAR: TipoDeficienciaFamiliarDef[] = [
  { valor: "CEGUEIRA", rotulo: "Cegueira" },
  { valor: "BAIXA_VISAO", rotulo: "Baixa Visão" },
  { valor: "SURDEZ_LEVE_MODERADA", rotulo: "Surdez leve/moderada" },
  { valor: "DEFICIENCIA_FISICA", rotulo: "Deficiencia Fisica" },
  {
    valor: "DEFICIENCIA_MENTAL_INTELECTUAL",
    rotulo: "Deficiencia Mental ou Intelectual",
  },
  { valor: "SINDROME_DOWN", rotulo: "Sindrome de Down" },
  { valor: "TRANSTORNO_MENTAL", rotulo: "Transtorno Mental" },
];

const porValor = new Map(TIPOS_DEFICIENCIA_FAMILIAR.map((t) => [t.valor, t]));

export function rotuloTipoDeficienciaFamiliar(
  valor: TipoDeficienciaFamiliar,
): string {
  return porValor.get(valor)?.rotulo ?? valor;
}
